File: mobile/lib/data/models/market.dart
/// Market index data model
class MarketIndex {
  final String name;
  final String price;
  final String change;
  final bool isUp;
  final String updatedAt;

  const MarketIndex({
    required this.name,
    required this.price,
    required this.change,
    required this.isUp,
    required this.updatedAt,
  });

  factory MarketIndex.fromJson(Map<String, dynamic> json) {
    return MarketIndex(
      name: json['name'] as String,
      price: json['price'] as String,
      change: json['change'] as String,
      isUp: json['isUp'] as bool? ?? false,
      updatedAt: json['updatedAt'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'price': price,
      'change': change,
      'isUp': isUp,
      'updatedAt': updatedAt,
    };
  }
}

/// Precious metal data model
class PreciousMetal {
  final String name;
  final double price;
  final double change;
  final String changeRate;
  final double open;
  final double high;
  final double low;
  final double close;
  final String unit;
  final String updatedAt;

  const PreciousMetal({
    required this.name,
    required this.price,
    required this.change,
    required this.changeRate,
    required this.open,
    required this.high,
    required this.low,
    required this.close,
    required this.unit,
    required this.updatedAt,
  });

  factory PreciousMetal.fromJson(Map<String, dynamic> json) {
    return PreciousMetal(
      name: json['name'] as String,
      price: (json['price'] as num).toDouble(),
      change: (json['change'] as num).toDouble(),
      changeRate: json['changeRate'] as String,
      open: (json['open'] as num).toDouble(),
      high: (json['high'] as num).toDouble(),
      low: (json['low'] as num).toDouble(),
      close: (json['close'] as num).toDouble(),
      unit: json['unit'] as String,
      updatedAt: json['updatedAt'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'price': price,
      'change': change,
      'changeRate': changeRate,
      'open': open,
      'high': high,
      'low': low,
      'close': close,
      'unit': unit,
      'updatedAt': updatedAt,
    };
  }
}

/// Gold price history data model
class GoldPrice {
  final String date;
  final String chinaGoldPrice;
  final String chowTaiFook;
  final String chinaChange;
  final String chowChange;

  const GoldPrice({
    required this.date,
    required this.chinaGoldPrice,
    required this.chowTaiFook,
    required this.chinaChange,
    required this.chowChange,
  });

  factory GoldPrice.fromJson(Map<String, dynamic> json) {
    return GoldPrice(
      date: json['date'] as String,
      chinaGoldPrice: json['chinaGoldPrice'] as String,
      chowTaiFook: json['chowTaiFook'] as String,
      chinaChange: json['chinaChange'] as String? ?? '',
      chowChange: json['chowChange'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'chinaGoldPrice': chinaGoldPrice,
      'chowTaiFook': chowTaiFook,
      'chinaChange': chinaChange,
      'chowChange': chowChange,
    };
  }
}

/// Volume trend data model
class VolumeTrend {
  final String date;
  final String totalVolume;
  final String shanghai;
  final String shenzhen;
  final String beijing;

  const VolumeTrend({
    required this.date,
    required this.totalVolume,
    required this.shanghai,
    required this.shenzhen,
    required this.beijing,
  });

  factory VolumeTrend.fromJson(Map<String, dynamic> json) {
    return VolumeTrend(
      date: json['date'] as String,
      totalVolume: json['totalVolume'] as String,
      shanghai: json['shanghai'] as String,
      shenzhen: json['shenzhen'] as String,
      beijing: json['beijing'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'totalVolume': totalVolume,
      'shanghai': shanghai,
      'shenzhen': shenzhen,
      'beijing': beijing,
    };
  }
}

/// Minute data model
class MinuteData {
  final String time;
  final String price;
  final String change;
  final String changeRate;
  final String volume;
  final String amount;

  const MinuteData({
    required this.time,
    required this.price,
    required this.change,
    required this.changeRate,
    required this.volume,
    required this.amount,
  });

  factory MinuteData.fromJson(Map<String, dynamic> json) {
    return MinuteData(
      time: json['time'] as String,
      price: json['price'] as String,
      change: json['change'] as String,
      changeRate: json['changeRate'] as String,
      volume: json['volume'] as String,
      amount: json['amount'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'time': time,
      'price': price,
      'change': change,
      'changeRate': changeRate,
      'volume': volume,
      'amount': amount,
    };
  }
}

File: backend/internal/model/market.go
package model

// ChangeStatus 涨跌状态
type ChangeStatus int

const (
	StatusDown ChangeStatus = -1 // 下跌
	StatusFlat ChangeStatus = 0  // 持平
	StatusUp   ChangeStatus = 1  // 上涨
)

// MarketIndex 市场指数
type MarketIndex struct {
	Name      string `json:"name"`
	Price     string `json:"price"`
	Change    string `json:"change"`
	IsUp      bool   `json:"isUp"`
	UpdatedAt string `json:"updatedAt"`
}

// PreciousMetal 贵金属
type PreciousMetal struct {
	Name       string  `json:"name"`
	Price      float64 `json:"price"`
	Change     float64 `json:"change"`
	ChangeRate string  `json:"changeRate"`
	Open       float64 `json:"open"`
	High       float64 `json:"high"`
	Low        float64 `json:"low"`
	Close      float64 `json:"close"`
	Unit       string  `json:"unit"`
	UpdatedAt  string  `json:"updatedAt"`
}

// GoldPrice 历史金价
type GoldPrice struct {
	Date           string `json:"date"`
	ChinaGoldPrice string `json:"chinaGoldPrice"`
	ChowTaiFook    string `json:"chowTaiFook"`
	ChinaChange    string `json:"chinaChange"`
	ChowChange     string `json:"chowChange"`
}

// VolumeTrend 成交量趋势
type VolumeTrend struct {
	Date        string `json:"date"`
	TotalVolume string `json:"totalVolume"`
	Shanghai    string `json:"shanghai"`
	Shenzhen    string `json:"shenzhen"`
	Beijing     string `json:"beijing"`
}

// MinuteData 分时数据
type MinuteData struct {
	Time       string `json:"time"`
	Price      string `json:"price"`
	Change     string `json:"change"`
	ChangeRate string `json:"changeRate"`
	Volume     string `json:"volume"`
	Amount     string `json:"amount"`
}

// Sector 行业板块
type Sector struct {
	ID               string `json:"id"`
	Name             string `json:"name"`
	ChangeRate       string `json:"changeRate"`
	MainNetInflow    string `json:"mainNetInflow"`
	MainInflowRatio  string `json:"mainInflowRatio"`
	SmallNetInflow   string `json:"smallNetInflow"`
	SmallInflowRatio string `json:"smallInflowRatio"`
}

// SectorFund 板块基金
type SectorFund struct {
	Code       string `json:"code"`
	Name       string `json:"name"`
	Type       string `json:"type"`
	Date       string `json:"date"`
	NetValue   string `json:"netValue"`
	Week1      string `json:"week1"`
	Month1     string `json:"month1"`
	Month3     string `json:"month3"`
	Month6     string `json:"month6"`
	YearToDate string `json:"yearToDate"`
	Year1      string `json:"year1"`
	Year2      string `json:"year2"`
	Year3      string `json:"year3"`
	SinceStart string `json:"sinceStart"`
}

// NewsItem 快讯
type NewsItem struct {
	ID          string       `json:"id"`
	Title       string       `json:"title"`
	Content     string       `json:"content"`
	Evaluate    string       `json:"evaluate"` // 利好/利空/空
	PublishTime int64        `json:"publishTime"`
	Entities    []NewsEntity `json:"entities"`
}

// NewsEntity 快讯关联股票
type NewsEntity struct {
	Code  string `json:"code"`
	Name  string `json:"name"`
	Ratio string `json:"ratio"`
}

// FundInfo 基金信息
type FundInfo struct {
	Code    string   `json:"code"`
	Name    string   `json:"name"`
	FundKey string   `json:"fundKey"`
	IsHold  bool     `json:"isHold"`
	Sectors []string `json:"sectors"`
}

// FundValuation 基金估值
type FundValuation struct {
	Code              string `json:"code"`
	Name              string `json:"name"`
	ValuationTime     string `json:"valuationTime"`
	Valuation         string `json:"valuation"`
	DayGrowth         string `json:"dayGrowth"`
	ConsecutiveDays   int    `json:"consecutiveDays"`
	ConsecutiveGrowth string `json:"consecutiveGrowth"`
	MonthlyStats      string `json:"monthlyStats"`
	MonthlyGrowth     string `json:"monthlyGrowth"`
}

// FundPoint 基金历史数据点
type FundPoint struct {
	Date  string `json:"date"`
	Value string `json:"value"`
}
